js/ui/camera.js

```diff
@@ -435,6 +435,10 @@ util.extend(Camera.prototype, /** @lends Map.prototype */{
         this.rotating = (startBearing !== bearing);
         this.pitching = (pitch !== startPitch);
 
+        if (options.smoothEasing && options.duration !== 0) {
+            options.easing = this._smoothOutEasing(options.duration);
+        }
```

Comment:
I think it's unnecessary — it only makes sense when there is a fast sequence of easings with varying intensities (the only use case for now is mouse wheel zoom), and the algorithm is only suited for easing with the default bezier curve.